### PR TITLE
Fix SDK names in the native SDKs

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Change SDK names of the Native SDKs when embedded in a Hybrid SDK.
_#skip-changelog_

## :bulb: Motivation and Context
Fix https://github.com/getsentry/sentry-dart/issues/1185

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [X] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [X] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: flutter/lib/src/version.dart [hunk 1]
```diff
@@ -2,4 +2,10 @@
...
 
 /// The default SDK name reported to Sentry.io in the submitted events.
-const String sdkName = 'sentry.dart.flutter';
+String sdkName(bool isWeb) => isWeb ? _browserSdkName : _ioSdkName;
+
+/// The default SDK name reported to Sentry.io in the submitted events.
+const String _ioSdkName = 'sentry.dart.flutter';
+
+/// The default SDK name reported to Sentry.io in the submitted events.
+const String _browserSdkName = '$_ioSdkName.browser';
```
Comment (marandaneto) [inline flutter/lib/src/version.dart]: I wonder if we want this level of granularity.

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt [hunk 2]
```diff
@@ -165,16 +165,19 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
...
-      args.getIfNotNull<Map<String, Any>>("sdk") { sdk ->
-        val name = sdk["name"] as? String
-        val version = sdk["version"] as? String
-        if (name != null && version != null) {
-          val sdkVersion = SdkVersion(name, version)
-          options.setSentryClientName("$name/$version")
-          options.setSdkVersion(sdkVersion)
-        }
+      val name = "sentry.java.android.flutter"
+      val version = io.sentry.android.core.BuildConfig.VERSION_NAME
```
Comment (marandaneto) [inline flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt]: Question, should we use the Native SDK's version (Android in this case), or the Flutter SDK version?
I believe the Native SDK version would make more sense/easier here.

**Comment by marandaneto** [comment]
@bruno-garcia can you double-check if the SDK names are matching your spec?

**Review by bruno-garcia — COMMENTED** [review]
The names make sense to me: basically appending just `.flutter` 👌

File: flutter/lib/src/version.dart (hunk 1 above)
Comment (marandaneto) [reply flutter/lib/src/version.dart]: We can break down the adoption using other tags instead of SDK name.

**Review by brustolin — APPROVED** [review]

**Pull request merged**